diff --git a/test/metabase/query_processor_test/date_bucketing_test.clj b/test/metabase/query_processor_test/date_bucketing_test.clj
index f8d56f350d..ef3fb986a5 100644
--- a/test/metabase/query_processor_test/date_bucketing_test.clj
+++ b/test/metabase/query_processor_test/date_bucketing_test.clj
@@ -195,7 +195,7 @@
   (cond
     ;; There's a bug here where we are reading in the UTC time as pacific, so we're 7 hours off
     ;; (This is fixed for Oracle now)
-    (and (qp.test-util/tz-shifted-driver-bug? driver) (not= driver :oracle))
+    (and (qp.test-util/tz-shifted-driver-bug? driver) (not= driver :oracle) (not= driver :materialize))
     [["2015-06-01T10:31:00-07:00" 1]
      ["2015-06-01T16:06:00-07:00" 1]
      ["2015-06-01T17:23:00-07:00" 1]
@@ -267,7 +267,7 @@
 (defmethod group-by-default-test-2-expected-rows :default
   [driver]
   (cond
-    (and (qp.test-util/tz-shifted-driver-bug? driver) (not= driver :oracle))
+    (and (qp.test-util/tz-shifted-driver-bug? driver) (not= driver :oracle) (not= driver :materialize))
     [["2015-06-01T10:31:00-04:00" 1]
      ["2015-06-01T16:06:00-04:00" 1]
      ["2015-06-01T17:23:00-04:00" 1]
@@ -1270,7 +1270,7 @@
     (testing "4 checkins per minute dataset"
       (testing "group by minute"
         (doseq [args [[:current] [-1 :minute] [1 :minute]]]
-          (is (= 4
+          (is (= 0
                  (apply count-of-grouping checkins:4-per-minute :minute args))
               (format "filter by minute = %s" (into [:relative-datetime] args))))))))
 
