diff --git a/test/metabase/query_processor_test.clj b/test/metabase/query_processor_test.clj
index b26ac03b83..fd5c2abb99 100644
--- a/test/metabase/query_processor_test.clj
+++ b/test/metabase/query_processor_test.clj
@@ -365,7 +365,7 @@
   also have this issue."
   [driver]
   ;; TIMEZONE FIXME — remove this and fix the drivers
-  (contains? #{:snowflake :oracle :redshift} driver))
+  (contains? #{:snowflake :oracle :redshift :materialize} driver))

 (defn nest-query
   "Nest an MBQL/native query by `n-levels`. Useful for testing how nested queries behave."
diff --git a/test/metabase/query_processor_test/explicit_joins_test.clj b/test/metabase/query_processor_test/explicit_joins_test.clj
index 901376afb1..350b51c5ef 100644
--- a/test/metabase/query_processor_test/explicit_joins_test.clj
+++ b/test/metabase/query_processor_test/explicit_joins_test.clj
@@ -265,8 +265,8 @@

 (deftest select-*-source-query-test
   (mt/test-drivers (disj (mt/normal-drivers-with-feature :left-join)
-                         ;; mongodb doesn't support foreign keys required by this test
-                         :mongo)
+                         ;; mongodb and materialize don't support foreign keys required by this test
+                         :mongo :materialize)
     (testing "We should be able to run a query that for whatever reason ends up with a `SELECT *` for the source query"
       (let [{:keys [rows columns]} (mt/format-rows-by [int int]
                                      (mt/rows+column-names
diff --git a/test/metabase/query_processor_test/date_bucketing_test.clj b/test/metabase/query_processor_test/date_bucketing_test.clj
index 87d225f5ae..343eb1f5f5 100644
--- a/test/metabase/query_processor_test/date_bucketing_test.clj
+++ b/test/metabase/query_processor_test/date_bucketing_test.clj
@@ -177,7 +177,7 @@

                ;; There's a bug here where we are reading in the UTC time as pacific, so we're 7 hours off
                ;; (This is fixed for Oracle now)
-               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle))
+               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle) (not= driver/*driver* :materialize))
                [["2015-06-01T10:31:00-07:00" 1]
                 ["2015-06-01T16:06:00-07:00" 1]
                 ["2015-06-01T17:23:00-07:00" 1]
@@ -232,7 +232,7 @@
                 ["2015-06-02 08:20:00" 1]
                 ["2015-06-02 11:11:00" 1]]

-               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle))
+               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle) (not= driver/*driver* :materialize))
                [["2015-06-01T10:31:00-04:00" 1]
                 ["2015-06-01T16:06:00-04:00" 1]
                 ["2015-06-01T17:23:00-04:00" 1]
diff --git a/test/metabase/query_processor/middleware/parameters/mbql_test.clj b/test/metabase/query_processor/middleware/parameters/mbql_test.clj
index cebabc72dc..e1f77fdb9a 100644
--- a/test/metabase/query_processor/middleware/parameters/mbql_test.clj
+++ b/test/metabase/query_processor/middleware/parameters/mbql_test.clj
@@ -287,7 +287,7 @@

 ;;
 (deftest handle-fk-forms-test
-  (mt/test-drivers (filter #(driver/supports? % :foreign-keys) (params-test-drivers))
+  (mt/test-drivers (filter #(driver/supports? % :foreign-keys) (params-test-drivers) (not= driver/*driver* :materialize))
     (testing "Make sure we properly handle paramters that have `fk->` forms in `:dimension` targets (#9017)"
       (is (= [[31 "Bludso's BBQ" 5 33.8894 -118.207 2]
               [32 "Boneyard Bistro" 5 34.1477 -118.428 3]
