diff --git a/test/metabase/query_processor_test/explicit_joins_test.clj b/test/metabase/query_processor_test/explicit_joins_test.clj
index 166598c4c6..e1bef679f1 100644
--- a/test/metabase/query_processor_test/explicit_joins_test.clj
+++ b/test/metabase/query_processor_test/explicit_joins_test.clj
@@ -262,8 +262,8 @@

 (deftest select-*-source-query-test
   (mt/test-drivers (disj (mt/normal-drivers-with-feature :left-join)
-                         ;; mongodb doesn't support foreign keys required by this test
-                         :mongo)
+                         ;; mongodb and materialize don't support foreign keys required by this test
+                         :mongo :materialize)
     (testing "We should be able to run a query that for whatever reason ends up with a `SELECT *` for the source query"
       (let [{:keys [rows columns]} (mt/format-rows-by [int int]
                                      (mt/rows+column-names
@@ -910,6 +910,7 @@

 (deftest join-with-brakout-and-aggregation-expression
   (mt/test-drivers (mt/normal-drivers-with-feature :left-join)
+    (when (not= driver/*driver* :materialize)
     (mt/dataset sample-dataset
       (let [query (mt/mbql-query orders
                                  {:source-query {:source-table $$orders
@@ -928,4 +929,4 @@
                   ["Doohickey" "Balistreri-Ankunding" "2018-02-01T00:00:00Z" 315.36 3.1536]
                   ["Doohickey" "Balistreri-Ankunding" "2018-03-01T00:00:00Z" 315.36 3.1536]]
                  (mt/formatted-rows [str str str 2.0 4.0]
-                   (qp/process-query query)))))))))
+                   (qp/process-query query))))))))))
diff --git a/test/metabase/query_processor_test/date_bucketing_test.clj b/test/metabase/query_processor_test/date_bucketing_test.clj
index 87d225f5ae..343eb1f5f5 100644
--- a/test/metabase/query_processor_test/date_bucketing_test.clj
+++ b/test/metabase/query_processor_test/date_bucketing_test.clj
@@ -177,7 +177,7 @@

                ;; There's a bug here where we are reading in the UTC time as pacific, so we're 7 hours off
                ;; (This is fixed for Oracle now)
-               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle))
+               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle) (not= driver/*driver* :materialize))
                [["2015-06-01T10:31:00-07:00" 1]
                 ["2015-06-01T16:06:00-07:00" 1]
                 ["2015-06-01T17:23:00-07:00" 1]
@@ -232,7 +232,7 @@
                 ["2015-06-02 08:20:00" 1]
                 ["2015-06-02 11:11:00" 1]]

-               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle))
+               (and (qp.test/tz-shifted-driver-bug? driver/*driver*) (not= driver/*driver* :oracle) (not= driver/*driver* :materialize))
                [["2015-06-01T10:31:00-04:00" 1]
                 ["2015-06-01T16:06:00-04:00" 1]
                 ["2015-06-01T17:23:00-04:00" 1]
